### PR TITLE
Reject create process instance command if unable to subscribe to events

### PR DESCRIPTION
**Pull request by korthout**
## Description

This provides an initial implementation to reject the Create Process Instance command when it's unable to subscribe to events. It's implemented using a workaround for the lack of support for
- #9420

The workaround has some implications:
- an error record is written along with the rejection
- the process instance is blacklisted (although the process instance is not really created)
- an exception is logged as an error including its stacktrace
- the rejection message is rather technical

## Related issues

closes #9422 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by korthout** [comment]
Flaky test: io.camunda.zeebe.it.clustering.ReaderCloseTest.shouldDeleteCompactedSegmentsFiles

File: engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceProcessor.java [hunk 1]
```diff
@@ -471,8 +473,13 @@ private void createEventSubscriptions(
...
           elementInstanceKey, elementRecord, ProcessInstanceIntent.ELEMENT_ACTIVATED);
 
-      catchEventBehavior.subscribeToEvents(
-          bpmnElementContext, catchEventSupplier, sideEffectQueue, commandWriter);
+      final Either<Failure, ?> subscribedOrFailure =
+          catchEventBehavior.subscribeToEvents(
+              bpmnElementContext, catchEventSupplier, sideEffectQueue, commandWriter);
+
+      if (subscribedOrFailure.isLeft()) {
+        throw new UncheckedExecutionException(subscribedOrFailure.getLeft().getMessage());
```
Comment (saig0) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceProcessor.java:481]: :wrench: We could enrich the error message with more context information.

If the error is logged then it is not obvious that the log statement is caused by a failed event subscription. The error message contains only the reason for the failure (e.g. `failed to evaluate expression 'unknown_var': no variable found for name 'unknown_var'`) but doesn't mention that the processor tries to create an event subscription for a scope. Even, if it's visible in the stacktrace.

**Review by saig0 — APPROVED** [review]
@korthout looks good :+1: 

I have only one optional suggestion. Feel free to ignore if you don't agree.

**Review by saig0 — APPROVED** [review]
👍 LGTM

Bors merge

**Pull request merged**